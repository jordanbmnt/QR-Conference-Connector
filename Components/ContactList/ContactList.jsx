import { StyleSheet } from "react-native";
import { ScrollView } from "@gluestack-ui/themed";
import UserCard from "../UserCard/UserCard";

export default ContactList = ({ contacts }) => {
  return (
    <ScrollView height='100%' sx={styles.scrollView}>
      {contacts.map((user, id) => (
        <UserCard key={id} userInfo={user} />
      ))}
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  scrollView: {
    alignContent: "center",
  },
});
