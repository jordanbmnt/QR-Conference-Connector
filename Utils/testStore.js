import { render } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { setupUserInfoStore } from "../Redux/store";
import { GluestackUIStyledProvider } from "@gluestack-ui/themed";
import { config } from "@gluestack-ui/config";

export const initialState = {
  "appSlice": {
    viewContactInfo: false,
  scanViewVisible: true,
  previousPage: false,
  contactInfo: {},
  alertInfo: {
    visibility: false,
    theme: "",
  }
  },
};

export function renderWithProviders(
  ui,
  {
    preloadedState = initialState,
    store = setupUserInfoStore(preloadedState),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return (
      <Provider store={store}>
        <GluestackUIStyledProvider config={config}>
          {children}
        </GluestackUIStyledProvider>
      </Provider>
    );
  }
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}
