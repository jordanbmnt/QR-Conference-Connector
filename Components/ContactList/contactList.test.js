import ContactList from "./ContactList";
import { renderWithProviders } from "../../Utils/testStore";

const contactList = [
  {
    companyName: "",
    email: "jackie@gmail.com",
    github: "https://github.com/jackie",
    id: 1,
    name: "Jake",
    position: "CEO of Marketing",
    profilePic: "Image",
    site: "https://www.spotify.com",
    status: "unavailable",
    statusColor: "grey",
  },
  {
    companyName: "CN",
    email: "jake@adventuretime.com",
    github: "github.com/jake",
    id: 5,
    name: "Jake",
    position: "Full-Stack Web Developer",
    profilePic: "Image",
    site: "cartoon.com",
    status: "alien",
    statusColor: "purple",
  },
];

describe("ContactList", function () {
  let emailContactList;
  beforeEach(() => {
    emailContactList = renderWithProviders(
      <ContactList contacts={contactList} />
    );
  });
  it("should render the emailContactList component correctly", () => {
    expect(emailContactList).toMatchSnapshot();
  });
  it("should display a list of two contacts", function () {
    const contactListLength = emailContactList.getAllByText("Jake").length;
    expect(contactListLength).toBe(2);
  });
});
