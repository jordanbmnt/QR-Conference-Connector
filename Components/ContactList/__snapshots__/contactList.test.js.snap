// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactList should render the emailContactList component correctly 1`] = `
<RCTScrollView
  data-style=""
  dataSet={
    {
      "style": "",
    }
  }
  style={
    [
      undefined,
      {},
      {},
      {},
      {
        "alignContent": "center",
        "height": "100%",
      },
      undefined,
    ]
  }
  sx={
    {
      "alignContent": "center",
    }
  }
>
  <View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="pressable"
    >
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            {},
            {},
            {
              "borderColor": "#D4D4D4",
              "borderRadius": 8,
              "borderWidth": 1,
              "margin": 20,
              "paddingHorizontal": 36,
              "paddingVertical": 36,
            },
            undefined,
          ]
        }
        sx={
          {
            "_dark": {
              "borderColor": "$backgroundDark700",
            },
          }
        }
      >
        <View
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          space="2xl"
          style={
            [
              undefined,
              {},
              {
                "flexDirection": "column",
              },
              {},
              {
                "gap": 24,
              },
              undefined,
            ]
          }
        >
          <View
            data-style=""
            dataSet={
              {
                "style": "",
              }
            }
            space="md"
            style={
              [
                undefined,
                {},
                {
                  "flexDirection": "row",
                },
                {},
                {
                  "gap": 12,
                },
                undefined,
              ]
            }
          >
            <View
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              size="md"
              style={
                [
                  undefined,
                  {},
                  {
                    "alignItems": "center",
                    "backgroundColor": "#005DB4",
                    "borderRadius": 9999,
                    "justifyContent": "center",
                    "position": "relative",
                  },
                  {},
                  {
                    "height": 48,
                    "width": 48,
                  },
                  undefined,
                ]
              }
            >
              <Text
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                size="md"
                style={
                  [
                    undefined,
                    {},
                    {
                      "color": "#525252",
                      "fontFamily": undefined,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {},
                    {
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    undefined,
                    {},
                    {
                      "color": "#FCFCFC",
                      "fontWeight": "600",
                      "overflow": "hidden",
                      "textTransform": "uppercase",
                    },
                    {},
                    {},
                    undefined,
                  ]
                }
              >
                J
              </Text>
              <Image
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                onError={[Function]}
                source={
                  {
                    "uri": "Image",
                  }
                }
                style={
                  [
                    undefined,
                    {},
                    {
                      "borderRadius": 9999,
                      "height": "100%",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {},
                    undefined,
                  ]
                }
              />
            </View>
            <View
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              style={
                [
                  undefined,
                  {},
                  {
                    "flexDirection": "column",
                  },
                  {},
                  undefined,
                ]
              }
            >
              <Text
                accessibilityRole="header"
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  {
                    "color": "white",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontWeight": "700",
                    "letterSpacing": -0.2,
                    "lineHeight": 24,
                    "marginVertical": 0,
                  }
                }
              >
                Jake
              </Text>
              <View
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  [
                    undefined,
                    {},
                    {
                      "flexDirection": "row",
                    },
                    {},
                    {
                      "textJustify": "center",
                    },
                    undefined,
                  ]
                }
                sx={
                  {
                    "textJustify": "center",
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  style={
                    {
                      "color": "grey",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "letterSpacing": -0.2,
                      "lineHeight": 18,
                      "marginVertical": 0,
                    }
                  }
                >
                  unavailable
                </Text>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="16"
                  bbWidth="16"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  fill="none"
                  focusable={false}
                  height={16}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  role="image"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      [
                        undefined,
                        {},
                        {
                          "color": "#404040",
                        },
                        {},
                        {
                          "height": 16,
                          "width": 16,
                        },
                        undefined,
                        {},
                        {
                          "color": "#404040",
                        },
                        {},
                        {
                          "color": "grey",
                        },
                        undefined,
                      ],
                      {
                        "flex": 0,
                        "height": 16,
                        "width": 16,
                      },
                    ]
                  }
                  tintColor="grey"
                  vbHeight={24}
                  vbWidth={24}
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                    strokeWidth="2"
                  >
                    <RNSVGCircle
                      cx="12.1"
                      cy="12.1"
                      fill={null}
                      propList={
                        [
                          "fill",
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      r="1"
                      stroke={
                        {
                          "type": 2,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      testID="pressable"
    >
      <View
        data-style=""
        dataSet={
          {
            "style": "",
          }
        }
        style={
          [
            undefined,
            {},
            {},
            {},
            {
              "borderColor": "#D4D4D4",
              "borderRadius": 8,
              "borderWidth": 1,
              "margin": 20,
              "paddingHorizontal": 36,
              "paddingVertical": 36,
            },
            undefined,
          ]
        }
        sx={
          {
            "_dark": {
              "borderColor": "$backgroundDark700",
            },
          }
        }
      >
        <View
          data-style=""
          dataSet={
            {
              "style": "",
            }
          }
          space="2xl"
          style={
            [
              undefined,
              {},
              {
                "flexDirection": "column",
              },
              {},
              {
                "gap": 24,
              },
              undefined,
            ]
          }
        >
          <View
            data-style=""
            dataSet={
              {
                "style": "",
              }
            }
            space="md"
            style={
              [
                undefined,
                {},
                {
                  "flexDirection": "row",
                },
                {},
                {
                  "gap": 12,
                },
                undefined,
              ]
            }
          >
            <View
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              size="md"
              style={
                [
                  undefined,
                  {},
                  {
                    "alignItems": "center",
                    "backgroundColor": "#005DB4",
                    "borderRadius": 9999,
                    "justifyContent": "center",
                    "position": "relative",
                  },
                  {},
                  {
                    "height": 48,
                    "width": 48,
                  },
                  undefined,
                ]
              }
            >
              <Text
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                size="md"
                style={
                  [
                    undefined,
                    {},
                    {
                      "color": "#525252",
                      "fontFamily": undefined,
                      "fontStyle": "normal",
                      "fontWeight": "400",
                      "letterSpacing": 0,
                    },
                    {},
                    {
                      "fontSize": 16,
                      "lineHeight": 22,
                    },
                    undefined,
                    {},
                    {
                      "color": "#FCFCFC",
                      "fontWeight": "600",
                      "overflow": "hidden",
                      "textTransform": "uppercase",
                    },
                    {},
                    {},
                    undefined,
                  ]
                }
              >
                J
              </Text>
              <Image
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                onError={[Function]}
                source={
                  {
                    "uri": "Image",
                  }
                }
                style={
                  [
                    undefined,
                    {},
                    {
                      "borderRadius": 9999,
                      "height": "100%",
                      "position": "absolute",
                      "width": "100%",
                    },
                    {},
                    undefined,
                  ]
                }
              />
            </View>
            <View
              data-style=""
              dataSet={
                {
                  "style": "",
                }
              }
              style={
                [
                  undefined,
                  {},
                  {
                    "flexDirection": "column",
                  },
                  {},
                  undefined,
                ]
              }
            >
              <Text
                accessibilityRole="header"
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  {
                    "color": "white",
                    "fontFamily": undefined,
                    "fontSize": 18,
                    "fontWeight": "700",
                    "letterSpacing": -0.2,
                    "lineHeight": 24,
                    "marginVertical": 0,
                  }
                }
              >
                Jake
              </Text>
              <View
                data-style=""
                dataSet={
                  {
                    "style": "",
                  }
                }
                style={
                  [
                    undefined,
                    {},
                    {
                      "flexDirection": "row",
                    },
                    {},
                    {
                      "textJustify": "center",
                    },
                    undefined,
                  ]
                }
                sx={
                  {
                    "textJustify": "center",
                  }
                }
              >
                <Text
                  accessibilityRole="header"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  style={
                    {
                      "color": "grey",
                      "fontFamily": undefined,
                      "fontSize": 14,
                      "fontWeight": "700",
                      "letterSpacing": -0.2,
                      "lineHeight": 18,
                      "marginVertical": 0,
                    }
                  }
                >
                  alien
                </Text>
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="16"
                  bbWidth="16"
                  data-style=""
                  dataSet={
                    {
                      "style": "",
                    }
                  }
                  fill="none"
                  focusable={false}
                  height={16}
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  role="image"
                  stroke="currentColor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth={2}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      [
                        undefined,
                        {},
                        {
                          "color": "#404040",
                        },
                        {},
                        {
                          "height": 16,
                          "width": 16,
                        },
                        undefined,
                        {},
                        {
                          "color": "#404040",
                        },
                        {},
                        {
                          "color": "purple",
                        },
                        undefined,
                      ],
                      {
                        "flex": 0,
                        "height": 16,
                        "width": 16,
                      },
                    ]
                  }
                  tintColor="purple"
                  vbHeight={24}
                  vbWidth={24}
                  width={16}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={null}
                    propList={
                      [
                        "fill",
                        "stroke",
                        "strokeWidth",
                        "strokeLinecap",
                        "strokeLinejoin",
                      ]
                    }
                    stroke={
                      {
                        "type": 2,
                      }
                    }
                    strokeLinecap={1}
                    strokeLinejoin={1}
                    strokeWidth="2"
                  >
                    <RNSVGCircle
                      cx="12.1"
                      cy="12.1"
                      fill={null}
                      propList={
                        [
                          "fill",
                          "stroke",
                          "strokeWidth",
                          "strokeLinecap",
                          "strokeLinejoin",
                        ]
                      }
                      r="1"
                      stroke={
                        {
                          "type": 2,
                        }
                      }
                      strokeLinecap={1}
                      strokeLinejoin={1}
                      strokeWidth="2"
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
